lib/services/nodes-api-service.js

```diff
@@ -84,6 +84,8 @@ function nodeApiServiceFactory(
      * @return {Object}    node after removing relation
      */
     NodeApiService.prototype._removeRelation = function removeRelation(node, type, targets) {
+        var self = this;
+
         if (!node || !type || !_.has(node, 'relations')) {
             return;
```

Comment:
If you want to align all return as promise, this return and next one should return Promise.resolve.